### ISSUE TITLE
player.clan_rank returning NoneType

### ISSUE THREAD

**Issue by bobflem** [mathsman5133/coc.py#109]
### Example:
```
import coc
client = coc.login("email", "password")
async def main():
    clan = await client.get_clan('#8U9JL80J')
    async for player in clan.get_detailed_members():
        print(type(player.clan_rank))
client.loop.run_until_complete(main())
client.close()
```
returns
<class 'NoneType'> for every clan member

**Comment by majordoobie** [comment]
What happens when you print `player`

**Comment by bobflem** [comment]
That returns <class 'coc.players.Player'>

I'm able to return player.name and other player properties as well

**Comment by majordoobie** [comment]
Can you see if it works if you do the following:

```python
import coc
client = coc.login("email", "password")
async def main():
    clan = await client.get_clan('#8U9JL80J')
    for player in clan.members:
        print(type(player.clan_rank))

client.loop.run_until_complete(main())
client.close()

**Comment by Bla005-1** [comment]
Looking at the docs, I don't think you need to iterate through Clan.get_detailed_members, instead iterate through Clan.members. The difference is detailed members returns a Player object while members returns a ClanMember object.

**Comment by majordoobie** [comment]
> Looking at the docs, I don't think you need to iterate through Clan.get_detailed_members, instead iterate through Clan.members. The difference is detailed members returns a Player object while members returns a ClanMember object.

Yeah, just saw that. But it looks like it is intended to return the ClanMember if available but it fails.

**Comment by bobflem** [comment]
> Can you see if it works if you do the following:
> 
> ```python
> import coc
> client = coc.login("email", "password")
> async def main():
>     clan = await client.get_clan('#8U9JL80J')
>     for player in clan.members:
>         print(type(player.clan_rank))
> 
> client.loop.run_until_complete(main())
> client.close()
> ```

Returns ints! Makes sense, thanks for figuring it out

**Comment by majordoobie** [comment]
Awesome! Please keep the issue open so we can fix the bug.

**Comment by majordoobie** [comment]
Issue has been resolved with version v2.1.2
